apm-sniffer/apm-sdk-plugin/mysql-common/src/main/java/org/apache/skywalking/apm/plugin/jdbc/mysql/PreparedStatementExecuteMethodsInterceptor.java

```diff
@@ -40,15 +40,14 @@
     public final void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments,
                                    Class<?>[] argumentsTypes, MethodInterceptResult result) {
         StatementEnhanceInfos cacheObject = (StatementEnhanceInfos) objInst.getSkyWalkingDynamicField();
-        ConnectionInfo connectInfo = cacheObject.getConnectionInfo();
         /**
          * For avoid NPE. In this particular case, Execute sql inside the {@link com.mysql.jdbc.ConnectionImpl} constructor,
          * before the interceptor sets the connectionInfo.
-         *
+         * Fixed https://github.com/apache/skywalking/issues/4965. When invoking prepareCall, cacheObject is null. Because it will determine procedures's parameter types by querying the table of proc in mysql.
```

Comment:
Don't add the issue link, unless it is a very special thing.

Reply:
About `#prepareCall`, why `cacheObject` should be null? I prefer you should fix this NPE by having the correct ConnectionInfo rather than skipping it.

Reply:
You should forward the ConnectionInfo from Connection when `#prepareCall` called.

Reply:
When invoking prepareCall, it will determine procedures's parameter types by execute sql.` StatementEnhanceInfos` is set after prepareCall, so `cacheObject is` null in `PreparedStatementExecuteMethodsInterceptor.beforeMethod`. 
If we don't need to record the sql which determine procedures's parameter types when invoking prepareCall, we can skip it.